File: gin/models/product.go
package models

type Product struct {
	Id          string  `json:"id"`
	Name        string  `json:"name"`
	Description string  `json:"description"`
	UserId      string  `json:"userId"`
	Price       float64 `json:"price"`
	CategoryId  string  `json:"categoryId"`
}

File: aspnetcore/Models/Product.cs
namespace local_ly_dotnet.Models;

public class Product
{
    public required string Id { get; set; }
    public required string Name { get; set; }
    public required string Description { get; set; }
    public required string UserId { get; set; }
    public required decimal Price { get; set; }
    public required string CategoryId { get; set; }
}